qermit/taskgraph/mitres.py

```diff
@@ -302,3 +302,141 @@ def run(self, circuit_shots: List[CircuitShots]) -> List[BackendResult]:  # type
         :rtype: List[BackendResult]
         """
         return self([circuit_shots])[0]
+
+
+def split_shots_task_gen(max_shots: int) -> MitTask:
+    """When the number of shots requested is higher than max_shots, this task
+    splits the jobs into several individual jobs, where the total number of
+    shots matched that initially requested.
+
+    :param max_shots: The maximum number of shots per job.
+    :type max_shots: int
+    :return: A task dividing the job into several smaller ones.
+    :rtype: MitTask
+    """
+
+    def task(
+        obj, circuit_wires: List[CircuitShots]
+    ) -> Tuple[List[CircuitShots], List[int]]:
+        """This task divides a job into several smaller ones, if the number of
+        shots requested is larger then the maximum allowed.
+
+        :param circuit_wires: A list of the circuits to be run, and the number
+        of shots to be taken.
+        :type circuit_wires: List[CircuitShots]
+        :return: A new list of circuits, some of which repeat with a smaller
+        number of shots.
+        :rtype: Tuple[List[CircuitShots], List[int]]
+        """
+
+        # A new list of circuits and shots
+        split_circuit_wires = []
+        # and index of which circuits in the new list correspond to which in
+        # the original list. This is to say all circuits in the new list
+        # assigned an index i by this index list correspond to original
+        # circuit i.
+        split_index = []
+
+        # For each circuit, check if the number of shots requested is greater
+        # then the maximum, and divide the job into several smaller ones if
+        # this is the case.
+        for i, circ_shots in enumerate(circuit_wires):
+
+            shots_to_run = circ_shots.Shots
```

Comment:
Note that python can start doing some funny things with shots_to_run and circ_shots.Shots if you assign like this. I'd suggest replacing the whole function with this:
```

        # A new list of circuits and shots
        split_circuit_wires = []
        # and index of which circuits in the new list correspond to which in
        # the original list. This is to say all circuits in the new list
        # assigned an index i by this index list correspond to original
        # circuit i.
        split_index = []
        
        for i, c in enumerate(circuit_wires):
            div_val = c.Shots/max_shots
            if c.Shots > max_shots:
                split_circuit_wires.extend([CircuitShots(c.Circuit, max_shots) for _ in range(div_val)])
            split_circuit_wires.append(CircuitShots(c.Circuit, c.Shots % max_shots))
            split_index.extend(repeat(i, div_val + 1))

Reply:
i like, might want a ` div_val = c.Shots//max_shots` to be safe (floor divide)

Reply:
Great, thanks. Done now.